src/main/java/com/slack/kaldb/chunk/ChunkManager.java

```diff
@@ -263,18 +272,50 @@ public void rollOverActiveChunk() {
    *
    * TODO: Search chunks in parallel.
    */
-  public SearchResult<T> query(SearchQuery query) {
-    List<SearchResult<T>> searchResults = new ArrayList<>(chunkMap.size());
-    for (Chunk<T> chunk : chunkMap.values()) {
-      if (chunk.containsDataInTimeRange(
-          query.startTimeEpochMs / 1000, query.endTimeEpochMs / 1000)) {
-        LOG.debug("Searching chunk {}", chunk.id());
-        searchResults.add(chunk.query(query));
-      }
-    }
+  public CompletableFuture<SearchResult<T>> query(SearchQuery query) {
+
+    SearchResult<T> empty =
```

Comment:
`SearchResult` should probably provide a no-argument constructor instead.

Reply:
fixing this as part of #52